jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+number of ways. Including, but not limited to:
+
+=== Containment of credentials
+
+By using a single system of record for these builds, a service account, maintained by the 
+JENKINS-CERT team, can be used to access GitHub, deployment to Nexus, and deployment to 
+the update centers. Plugin maintainers need not leave their own credentials on a CI server 
+which they don't own, and permissions already in place in their GitHub repositories provide 
+the required controls over who can merge and release.
+
+=== Automatic enforcement of security best practices
+Autorelease builds will all come from `${trusted-ci-server}`. Rules 
+can be put in place on `${trusted-ci-server}`, which can provide implicit enforcement of the 
+Jenkins infrastructure team's security best practices. Compliance to these best practices becomes 
+something that plugin maintainers need not worry about.
+
+== Infrastructure Requirements
+
+We will need a number of things to get this going. The low level technical details will be described 
+in a separate Infrastructure Enhancement Request, so, this should be considered a summary for now:
+
+1. The webhook, configured on each participating plugin . Security implications of this are a bit beyond the scope of this document so far.
+2. A receiver for the aforementioned webhook, because the `${trusted-ci-server}` will be protected 
+behind a VPN
+3. A trusted Jenkins server for performing builds and deployments. 
+4. `buildPlugin` will need code added to validate `Autorelease.yaml` for correctness, and 
+build the plugin according to the settings described therein
+5. Service account(s), managed by the Jenkins Infrastructure team, which provide secure credentials 
+to systems such as Artifactory
+
+CAUTION: TODO: Things such as service accounts and new servers will be handled in a separate IEP.
+
+== Testing
+
+Autorelease brings with it a heightened importance for quality automated tests. However, there will be 
+no rules governing this. As is the case today, plugin maintainers are encouraged to release only 
+well-tested code, but there is little to stop someone from releasing something which is under-tested. 
+Autorelease does not change this in any way.
+
+== Prototype Implementation
+
+As a proof of concept, the (github-branch-source?) plugin will be the first to adopt. This provides 
+the initiative with a heavily used plugin, which sees relatively frequent releases already.
+
+A sample fork of github-branch-source could be provided as a reference implementation for 
+this proposal. It is understood that this need not be completed before this JEP is 
+"link:https://github.com/jenkinsci/jep/tree/master/jep/1#accepted[accepted]", but will need to 
+be made available before this JEP is given 
+"link:https://github.com/jenkinsci/jep/tree/master/jep/1#final[Final]" status.
+
+
+== References
+
+[[footnote-1]]1. Jenkins World 2017, link:http://bit.ly/2x1lCUZ[Contributor Summit Notes], pp. 11-12
```

Comment:
```suggestion
[[footnote-1]]1. Jenkins World 2017, link:[Contributor Summit Notes]
```

Reply:
If I make this change, we lose the ability for readers to find that document:

![image](https://user-images.githubusercontent.com/21689198/59638072-5c4edb00-9125-11e9-8472-0d6471623529.png)

Is that intentional?

Reply:
Did I not include a URL? I meant to. https://docs.google.com/document/d/1JSxYNI_RuA8ITlxVmxBdFg1A-sOKz-w7a9tzuPfWmr4/edit#heading=h.n2d90ci56ugy